tests/src/FunctionalJavascript/ContactSubmissionTest.php

```diff
@@ -0,0 +1,196 @@
+<?php
+
+namespace Drupal\Tests\webform_civicrm\FunctionalJavascript;
+
+use Behat\Mink\Exception\ElementNotFoundException;
+
+/**
+ * Tests submitting a Webform with CiviCRM and a single contact.
+ *
+ * @group webform_civicrm
+ */
+final class ContactSubmissionTest extends WebformCivicrmTestBase {
+
+  /**
+   * {@inheritdoc}
+   *
+   * During tests configuration schema is validated. This module does not
+   * provide schema definitions for its handler.
+   *
+   * To fix: webform.webform.civicrm_webform_test:handlers.webform_civicrm.settings
+   *
+   * @see \Drupal\Core\Test\TestSetupTrait::getConfigSchemaExclusions
+   */
+  protected static $configSchemaCheckerExclusions = [
+    'webform.webform.civicrm_webform_test',
+  ];
+
+  /**
+   * Test submitting a contact.
+   *
+   * @dataProvider dataContactValues
+   */
+  public function testSubmitWebform($contact_type, array $contact_values) {
+    $this->assertArrayHasKey('contact', $contact_values, 'Test data must contain contact');
+    $this->assertArrayHasKey('first_name', $contact_values['contact'], 'Test contact data must contain first_name');
+    $this->assertArrayHasKey('last_name', $contact_values['contact'], 'Test confact data must contain last_name');
+
+    // @todo change to login as admin then logout, then login as test user for submission
+    $this->drupalLogin($this->testUser);
```

Comment:
Currently `testUser` is an admin, and should be renamed as such. Then testUser should be a basic authenticated (or anonymous?) with permissions to submit the form.